### DISCUSSION TITLE
RTL support / Arabic ?

### DISCUSSION THREAD

**Discussion by mustafa-online** [sulu/sulu#7197] [Q&A]
I've just find sulu.
Does the admin panel supports **RTL** (right to left) ? if not, is there any plans?
I'm unable to find which CSS framework you working with.

**Comment by alexander-schranz** [comment]
Hello @mustafa-online,

we do not yet have RTL supported but we are highly interested what is the expected behaviour here to support it inside Sulu?

I think Sulu would require to detect if the current "Content Locale" is a RTL language depending on that render the `text_line`, `text_editor`, `text_area` with following CSS:

```css
    direction: rtl;
```

What other kind of things do we require to keep in mind, so Content can be added in the Arabic Language.

> I'm unable to find which CSS framework you working wit

Sulu does not use a specific CSS UI Framework, it has its own UI components build in React ontop of [CSS Modules](https://github.com/css-modules/css-modules).

**Reply by mustafa-online** [reply]
Thanks for your reponse, I'm interested in **RTL**ing the whole ui, because in arabic
We do everything from right to left. So which **css** files can I change to do so ?

**Reply by mustafa-online** [reply]
if we use the "**CSS Logical Properties**" we can achieve this easily.
I can open a PR, but I dont know which files to start from. there is no guide for this in Docs.

**Reply by alexander-schranz** [reply]
UI is sure more difficult here. I think the first part should concentrate on making the `Content Fields` RTL support.

For deeper understanding there are 2 kind of languages in Sulu the "System Language" and the "Content Language".
The Content Language is what is inside the input, textarea, .. fields and representing what is in the language chooser on the Top Right. The `System Language` is in which language the User Admin sees the UI. So it can be that a User itself is in the Admin as `English` (LTR) and so see all Labels and so in `English` but need to change `Arabic` content which should then be `RTL`. That is sure the first thing which need to be tackled that they can write RTL when the Content Language is Arabic or in another RTL language, but the UI itself is still English (LTR).

The Admin UI / System Language support for Arabic itself is a different Topic and should be targetted by its own as it is more difficult as there are lot of calculations and things done for some components like Tabs, Blocks, Toolbars and so would be effected CSS Logical Properties. That just changing to logical properties will make some calculation issues. So you probably really would require to go over every component and test it.

What I'm curious about how it effects the main components of the UI if we add RTL support. Is then the Main Navigation also on the Right side and the Preview on the Left side? Or will the main component stay but just be itself RTL. What are there common design practices here?

**Reply by mustafa-online** [reply]
> Is then the Main Navigation also on the Right side and the Preview on the Left side?

Yes, everything should be mirrored, check this screenshot from WordPress. inside the text editor we can use the Bidi button to
work on multi-ligual content.

![WordPress-Dashboard-1024x506](https://github.com/sulu/sulu/assets/5323832/5f2ec2ed-7570-4091-838e-7196e654f468)

**Reply by alexander-schranz** [reply]
How does wordpress target if I want to have my UI in English but have Arabic content in it? How does a form with input, textareas, look like that?

**Reply by mustafa-online** [reply]
The content is up to the user, when you change the admin language to Arabic the direction of components respect that accordingly.
<html lang="ar" dir="rtl">
It's not that complicated, each lang has it's direction.. the text editor can handle both langs at the same time.. the main issue is with layout direction.

**Reply by alexander-schranz** [reply]
Do you have a screenshot for mixed content. UI in wordpress English and the Content Language is Arabic or another RTL language?

**Reply by mustafa-online** [reply]
I don't have it right now, but i can find one for you.. 
but anyway: do you think it's impossibe for Sulu to add RTL for now?
I'm trying to be helpful here.
